src/meltano/cli/__init__.py

```diff
@@ -67,16 +91,19 @@ def _run_cli():
         KeyboardInterrupt: if caught.
     """
     try:
-        try:  # noqa: WPS505
+        try:  # noqa: WPS225, WPS505
             cli(obj={"project": None})
         except ProjectReadonly as err:
             raise CliError(
                 f"The requested action could not be completed: {err}"
             ) from err
         except KeyboardInterrupt:  # noqa: WPS329
             raise
+        except MeltanoError as err:
+            handle_meltano_error(err)
         except Exception as err:
-            raise CliError(str(err)) from err
+            message = f"{troubleshooting_message}\n{err}"
+            raise CliError(message) from err
```

Comment:
This isn't introduced by this PR, but this approach of turning `err` into a string loses the traceback, which can lead to some unhelpful error messages. The traceback is still kept in the `CliError` though, so as long as we log it somewhere it's fine. That said, I believe it is being logged.